packages/sui-js-compiler/index.js

```diff
@@ -6,17 +6,77 @@
 import program from 'commander'
 import fg from 'fast-glob'
 import fs from 'fs-extra'
+import path from 'node:path'
 
 import {transformFile} from '@swc/core'
 
 import {getSWCConfig} from './swc-config.js'
 
+const SOURCE_DIR = './src'
+const OUTPUT_DIR = './lib'
+const TS_EXTENSION_REGEX = /(\.ts)x?/
+const COMPILED_EXTENSION = '.js'
+const DEFAULT_TS_CONFIG = {
+  declaration: true,
+  emitDeclarationOnly: true,
+  incremental: true,
+  jsx: 'react-jsx',
+  module: 'es6',
+  esModuleInterop: true,
+  noImplicitAny: false,
+  baseUrl: '.',
+  outDir: OUTPUT_DIR,
+  skipLibCheck: true,
+  strict: true,
+  target: 'es5',
+  types: ['react', 'node']
+}
+
+// Get TS config from the package dir.
+// If present, set TypeScript as enabled.
+const tsConfigPath = path.join(process.cwd(), 'tsconfig.json')
+let tsConfigData
+let isTypeScriptEnabled = false
+
+try {
+  if (fs.existsSync(tsConfigPath)) {
+    tsConfigData = JSON.parse(fs.readFileSync(tsConfigPath, {encoding: 'utf8'}))
+    isTypeScriptEnabled = true
+  }
+} catch (err) {
+  console.error(err)
+}
+
 const compileFile = async (file, options) => {
   const {code} = await transformFile(file, getSWCConfig(options))
-  const outputPath = file.replace('./src', './lib')
+  const outputPath = file
+    .replace(SOURCE_DIR, OUTPUT_DIR)
+    .replace(TS_EXTENSION_REGEX, COMPILED_EXTENSION)
+
   fs.outputFile(outputPath, code)
 }
 
+const compileTypes = async (files, options) => {
+  const {createCompilerHost, createProgram} = await import('typescript').then(
+    module => module.default
+  )
```

Comment:
Why do you need to async import `typescript` ?

Reply:
Yep, in fact, this all is a dev tool, there is no a real improvement. I was a little bit confused cause we're defining all the deps as "production" `dependencies`...